internal/pkg/source/source.go

```diff
@@ -106,9 +106,10 @@ func (a *Source) visitReferrers(referrers *[]ssa.Instruction) {
 		case *ssa.Call:
 			// This is to avoid attaching calls where the source is the receiver, ex:
 			// core.Sinkf("Source id: %v", wrapper.Source.GetID())
-			if v.Call.Signature().Recv() != nil {
+			if recv := v.Call.Signature().Recv(); recv != nil && a.config.IsSource(utils.Dereference(recv.Type())) {
 				continue
```

Comment:
For the case `GetData()` rather than `GetID()`, are we explicitly superseding this elsewhere?

Reply:
Yeah, we're using the `fieldpropagators` analyzer's result in `levee.go:61`:
```
				case fieldPropagators.Contains(v):
					sources = append(sources, source.New(v, conf))
```

I think it would make more sense to use the `fieldpropagator` results in the `source` analyzer, WDYT?

Reply:
That sounds good to me.